lib/Parse/SyntaxParsingContext.cpp

```diff
@@ -297,7 +297,7 @@ void finalizeSourceFile(RootContextData &RootData,
   assert(newRaw);
   SF.setSyntaxRoot(make<SourceFileSyntax>(newRaw));
 
-  if (SF.getASTContext().LangOpts.VerifySyntaxTree) {
+  if (EOFToken->isPresent() && SF.getASTContext().LangOpts.VerifySyntaxTree) {
```

Comment:
Nice catch! Can we add some comments here why this is checked? If `finalizeSourceFile` gets called before we really finalize it, we may need to rename this function to a more descriptive one.